lib/pub_sub/poller.rb

```diff
@@ -1,31 +1,38 @@
 module PubSub
   class Poller
-    def initialize(queue_url, verbose = false)
-      @queue_url = queue_url
+    def initialize(verbose = false)
       @verbose = verbose
     end
 
     def poll
-      poller.poll(config) do |message|
-        if @verbose
-          PubSub.logger.info(
-            "PubSub received: #{message.message_id} - #{message.body}"
-          )
+      loop do
+        Breaker.current_breaker.run do
+          poller.poll(idle_timeout: 60) do |message|
+            if @verbose
+              PubSub.logger.info(
+                "PubSub received: #{message.message_id} - #{message.body}"
+              )
+            end
+            Message.new(message.body).process
+          end
         end
-        Message.new(message.body).process
+        Breaker.use_next_breaker
```

Comment:
Just trying to understand this.  Normally the `poller.poll` method will loop indefinitely, but if you pass the `idle_timeout` option, it'll stop after this many seconds, which will happen if there are no messages to be retrieved, at which point you'll start polling the next region, and so on.  So you'll be switching regions even if there isn't a problem on the given region.  This will also affect the publishing of messages, since once you switch to the next region, all published messages will also use this region.  Is this the desired behaviour?